codegen/scala-gen/src/main/scala/com/akkaserverless/codegen/scalasdk/impl/ValueEntityTestKitGenerator.scala

```diff
@@ -164,4 +166,72 @@ object ValueEntityTestKitGenerator {
           |""".stripMargin)
   }
 
+  def integrationTest(
+      main: FullyQualifiedName,
+      valueEntity: ModelBuilder.ValueEntity,
+      service: ModelBuilder.EntityService): File = {
+
+    val client = FullyQualifiedName(service.fqn.name + "Client", service.fqn.parent)
+
+    implicit val imports: Imports =
+      generateImports(
+        Seq(main, valueEntity.state.fqn, client) ++
+        service.commands.map(_.inputType) ++
+        service.commands.map(_.outputType),
+        service.fqn.parent.scalaPackage,
+        otherImports = Seq(
+          "com.akkaserverless.scalasdk.valueentity.ValueEntity",
+          "com.akkaserverless.scalasdk.testkit.ValueEntityResult",
+          "com.akkaserverless.scalasdk.testkit.AkkaServerlessTestkit",
+          "org.scalatest.matchers.should.Matchers",
+          "org.scalatest.wordspec.AnyWordSpec",
+          "org.scalatest.BeforeAndAfterAll",
+          "org.scalatest.concurrent.ScalaFutures",
+          "org.scalatest.time.Span",
+          "org.scalatest.time.Seconds",
+          "org.scalatest.time.Millis"),
+        packageImports = Seq(valueEntity.fqn.parent.scalaPackage),
+        semi = false)
+
+    val entityClassName = service.fqn.name
+
+    File(
+      service.fqn.fileBasename + "IntegrationSpec.scala",
```

Comment:
if we have these in test scope I wonder if there is a way to run all unit tests without these?

Thinking about `testOnly **Spec` but that would include IntegrationSpec too. Shall we name the others `UnitSpec` so that `testOnly **UnitSpec` would work?

Reply:
`*UnitSpec` could work, though it's a bit unfortunate when ideally you'd want to have many unit tests and few integration tests. The only alternative I can think of is using ScalaTest tagging, though that's a bit obscure.

Reply:
Created issue https://github.com/lightbend/akkaserverless-java-sdk/issues/571 for this thought